app/src/main/java/org/kiwix/kiwixmobile/local_file_transfer/FileItem.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+@file:Suppress("PackageNaming")
+
+package org.kiwix.kiwixmobile.local_file_transfer
+
+import android.net.Uri
+import androidx.annotation.IntDef
+import org.kiwix.kiwixmobile.local_file_transfer.WifiDirectManager.Companion.getFileName
+
+/**
+ * Helper class, part of the local file sharing module.
+ *
+ * Defines a file-item to represent the files being transferred.
+ */
+class FileItem private constructor(val fileUri: Uri?, val fileName: String) {
+
+  var fileStatus = FileStatus.TO_BE_SENT
+
+  @Retention(AnnotationRetention.SOURCE)
+  @IntDef(FileStatus.TO_BE_SENT, FileStatus.SENDING, FileStatus.SENT, FileStatus.ERROR)
+  annotation class FileStatus {
```

Comment:
We could finally migrate this to an enum

Reply:
Done👍

Reply:
This is not an enum. [docs](https://kotlinlang.org/docs/reference/enum-classes.html)

Reply:
Judging from the other comments maybe you just didn't push your changes.

It might be best to let me mark things as resolved as they should only be marked that way if the comment has been resolved and committed

Reply:
@macgills Yes, i am facing errors related to kotlin-extensions.

Reply:
You may have to add an import for Viewgroup.inflate manually? IIRC kotlin 1.3.71 release was the ne that fixed the dodgy import on object scoped extension functions

Reply:
Have added the import, and pushed the commit with --no-verify for now, please have a look.

Reply:
Ah, you should move this stuff into the `ViewHolder.bind` method, that should resolve it

Reply:
and ` BaseViewHolder<View>` should be ` BaseViewHolder<FileItem>`